lib/tenstorrent/bh_arc/wipe_l1.c

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (c) 2024 Tenstorrent AI ULC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "eth.h"
+#include "gddr.h"
+#include "harvesting.h"
+#include "init.h"
+#include "noc.h"
+#include "noc2axi.h"
+#include "noc_dma.h"
+
+#include <tenstorrent/sys_init_defines.h>
+
+#define ERISC_L1_SIZE  (512 * 1024)
+#define MRISC_L1_SIZE  (128 * 1024)
+#define TENSIX_L1_SIZE (1536 * 1024)
+
+#define ARC_NOC0_X 8
+#define ARC_NOC0_Y 0
+
+extern uint8_t large_sram_buffer[SCRATCHPAD_SIZE] __aligned(4);
+
+/* use any arbitrary non-harvested tensix core as the source for wiping l1s */
+static const uint8_t tensix_x = 1;
+static const uint8_t tensix_y = 2;
+
+/* this function first clears an arbitrary non-harvested tensix core and do a NOC DMA broadcast to
+ * clear all remaining tensix l1s
+ */
+static void wipe_tensix_l1(void)
```

Comment:
This really should be done in a file (ideally a driver) that is dedicated to initializing tensix.